ride-service/src/main/java/com/modsen/rideservice/service/impl/RideServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+        checkExistence(id);
+        rideRepository.deleteById(id);
+    }
+
+    public RideResponse getById(Long id) {
+        return toDTO(getEntityById(id));
+    }
+
+    public RideResponse addRide(RideCreationRequest dto) {
+        Ride ride = toModel(dto);
+        ride.setInitialCost(Math.round(new Random().nextDouble(5, 20) * 100) / 100.0);
+        setFinalCost(ride, dto.getPromoCode());
+        ride.setDate(new Date());
+        ride.setStatus(Status.NOT_ACCEPTED);
+        ride.setDriverId(getAvailableDriverId(null));
+        ride.setStatus(Status.ACCEPTED);
+        return toDTO(rideRepository.save(ride));
+    }
+
+    private void setFinalCost(Ride ride, String promoCodeName) {
+        if (!promoCodeName.isEmpty()) {
+            Double coefficient = promoCodeService.getById(promoCodeName).getCoefficient();
+            ride.setFinalCost(Math.round(ride.getInitialCost() * coefficient * 100.0) / 100.0);
+        } else ride.setFinalCost(ride.getInitialCost());
+    }
+
+    private Ride getEntityById(Long id) {
+        Optional<Ride> ride = rideRepository.findById(id);
+        if (ride.isPresent()) return ride.get();
+        throw new NotFoundException("id", "Ride with id={" + id + "} not found");
+    }
+
+    private Long getAvailableDriverId(DriverRejectRequest dto) {
+        //здесь будет отправка запроса на сервис водителей для получения свободного
+        //временно реализовано рандомное генерирование id водителя
+        return new Random().nextLong(1, 10);
+    }
+
+    public void finishRide(Long id, DriverFinishRequest dto) {
+        Ride ride = getEntityById(id);
+        if (ride.getStatus() != Status.STARTED)
+            throw new WrongStatusException("status", "Ride with id={" + id + "} not started");
+        ride.setStatus(Status.FINISHED);
+        Rating rating = Rating.builder()
+                .ride(ride)
+                .passengerRating(dto.getRatingToPassenger())
+                .build();
+        ride.setRating(rating);
+        rideRepository.save(ride);
+    }
+
+    public void finishRide(Long id, PassengerFinishRequest dto) {
+        Ride ride = getEntityById(id);
+        if (dto.getRatingToDriver() != null) ride.getRating().setDriverRating(dto.getRatingToDriver());
+        rideRepository.save(ride);
+    }
+
+    public void startRide(Long id, UserIdRequest dto) {
+        Ride ride = getEntityById(id);
+        checkDriverAccess(ride, dto.getUserId());
+        if (ride.getStatus() != Status.ACCEPTED)
+            throw new WrongStatusException("status", "Ride with id={" + id + "} has wrong status");
+        ride.setStatus(Status.STARTED);
+        rideRepository.save(ride);
+    }
+
+    public void passengerRejectRide(Long id, UserIdRequest dto) {
+        Ride ride = getEntityById(id);
+        checkPassengerAccess(ride, dto.getUserId());
+        checkPossibilityToReject(ride);
+        ride.setStatus(Status.REJECTED);
+        rideRepository.save(ride);
```

Comment:
Разделяй пустыми строками логические блоки, так код становится понятнее
```suggestion
        Ride ride = getEntityById(id);
        
        checkPassengerAccess(ride, dto.getUserId());
        checkPossibilityToReject(ride);
        
        ride.setStatus(Status.REJECTED);
        rideRepository.save(ride);
```